v2/coord/behaviour_test.go

```diff
@@ -0,0 +1,28 @@
+package coord
+
+import (
+	"context"
+)
+
+type NullSM[E any, S any] struct{}
```

Comment:
I think I prefer to fully type `StateMachine` instead of just `SM` - although we're almost entering Java land with that :D